### ISSUE TITLE
graphql 0.13.*

### ISSUE THREAD

**Issue by lifeiscontent** [graphql/graphql-language-service#219]
Any ideas when the graphql 0.13.* will be supported?

**Comment by barbalex** [comment]
This is getting very important.

I am getting an increasing number of problems due to different versions of graphql in the codebase.

And most of all: due to an issue in graphql versions before v13 that creates errors when multiple instances of graphql (even the same version) were created, see: https://github.com/graphql/graphql-js/releases/tag/v0.13.0 "Only trigger "one instance of graphql" error in DEV environments, removing false errors from minified bundles".

For instance: Since updating postgraphile-plugin-connection-filter from v1.0.0-beta.4 to 1.0.0-beta.5 my app does not run any more due to warnings on the server:

```
  graphile-build:warn Ensure that there is only one instance of "graphql" in the node_modules
  graphile-build:warn directory. If different versions of "graphql" are the dependencies of other
  graphile-build:warn relied on modules, use "resolutions" to ensure only one version is installed.
```

and errors the server then returns on queries to the app (with the exact same message).

And this happens even thought I have added a resolutions key to package.json:
```js
"resolutions": {
    "graphql": "0.13.0"
}
```

`npm ls` graphql lists a big number of occurrences of graphql in my codebase due to the tools used (apollo and postgraphile stack):
```
alexa@DESKTOP-6C4V66Q C:\Users\alexa\ae2
> npm ls graphql
ae2@0.1.1 C:\Users\alexa\ae2
+-- apollo-cache-inmemory@1.1.12
| `-- graphql@0.13.0  extraneous
+-- apollo-client@2.2.8
| `-- graphql@0.13.0  extraneous
+-- apollo-link@1.2.1
| `-- graphql@0.13.0  extraneous
+-- apollo-link-http@1.5.3
| `-- graphql@0.13.0  extraneous
+-- graphiql@0.11.11
| `-- codemirror-graphql@0.6.12
|   `-- graphql-language-service-interface@1.0.18
|     `-- graphql-config@1.1.4
|       +-- graphql@0.13.0  invalid
|       `-- graphql-import@0.1.9
|         `-- graphql@0.13.0  invalid
+-- graphql-tag@2.8.0
| `-- graphql@0.13.0  extraneous
+-- graphql-voyager@1.0.0-rc.15
| `-- graphql@0.13.0  extraneous
`-- postgraphile@4.0.0-beta.5
  +-- graphql@0.13.0  invalid
  `-- postgraphile-core@4.0.0-beta.2
    +-- graphile-build@4.0.0-beta.2
    | `-- graphql@0.13.0  extraneous
    `-- graphile-build-pg@4.0.0-beta.2
      `-- graphql-iso-date@3.5.0
        `-- graphql@0.13.0  extraneous

npm ERR! extraneous: graphql@0.13.0 C:\Users\alexa\ae2\node_modules\apollo-cache-inmemory\node_modules\graphql
npm ERR! extraneous: graphql@0.13.0 C:\Users\alexa\ae2\node_modules\apollo-client\node_modules\graphql
npm ERR! extraneous: graphql@0.13.0 C:\Users\alexa\ae2\node_modules\apollo-link\node_modules\graphql
npm ERR! extraneous: graphql@0.13.0 C:\Users\alexa\ae2\node_modules\apollo-link-http\node_modules\graphql
npm ERR! invalid: graphql@0.13.0 C:\Users\alexa\ae2\node_modules\graphql-config\node_modules\graphql
npm ERR! invalid: graphql@0.13.0 C:\Users\alexa\ae2\node_modules\graphql-import\node_modules\graphql
npm ERR! extraneous: graphql@0.13.0 C:\Users\alexa\ae2\node_modules\graphql-tag\node_modules\graphql
npm ERR! extraneous: graphql@0.13.0 C:\Users\alexa\ae2\node_modules\graphql-voyager\node_modules\graphql
npm ERR! invalid: graphql@0.13.0 C:\Users\alexa\ae2\node_modules\postgraphile\node_modules\graphql
npm ERR! extraneous: graphql@0.13.0 C:\Users\alexa\ae2\node_modules\graphile-build\node_modules\graphql
npm ERR! extraneous: graphql@0.13.0 C:\Users\alexa\ae2\node_modules\graphql-iso-date\node_modules\graphql
```

Someone else has asked postgraphile to update to graphql v13 but that seems blocked because of codemirror-graphql, which in turn depends on graphql-language-service: https://github.com/graphile/postgraphile/issues/720.

Seems that graphql-import@0.1.9 is one of the reasons graphql v13 can't be used. The up to date version (0.5.0) enables that though: https://github.com/graphcool/graphql-import/blob/master/package.json

So updating graphql-import could help.

Same is true for graphql-config@1.1.4 (up to date: v2.0.1): https://github.com/graphcool/graphql-config/blob/master/package.json

**Comment by james-ff** [comment]
+1

**Comment by lifeiscontent** [comment]
welp, I guess GraphQL is dead. Long live SQL 😝

**Comment by james-ff** [comment]
I'm in dependency hell 3 libraries up from this

**Comment by TSMMark** [comment]
I got so tired of NPM peerDependency hell I wrote this snippet to recursively remove all `peerDependency` records from all `package.json` files in `node_modules`. This is probably a terrible idea but oh well it let me write my shrinkwrap/lockfile sooooo... Enjoy!

https://gist.github.com/TSMMark/5488758028b10d537bf1f721198a539e

**Comment by lifeiscontent** [comment]
@gaearon do you know anyone who could get the ball rolling on this issue?

**Comment by lostplan** [comment]
v1.2.2 is released with support for graphql@0.13 peer-dependency. Let me know how you get on.
